### ISSUE TITLE
Uncaught SyntaxError: Unexpected token /

### ISSUE THREAD

**Issue by perfecto25** [icetee/remote-ftp#213]
[Enter steps to reproduce below:]
1. ...
2. ...

**Atom Version**: 1.0.19
**System**: Linux Mint
**Thrown From**: [Remote-FTP](https://github.com/mgrenier/remote-ftp) package, v0.7.8
### Stack Trace

Uncaught SyntaxError: Unexpected token /

```
At file:///opt/atom/resources/app.asar/static/index.html#%7B%22locationsToOpen%22%3A%5B%7B%22pathToOpen%22%3A%22%2Fhome%2Fmrx%2FDownloads%22%2C%22initialLine%22%3Anull%2C%22initialColumn%22%3Anull%7D%5D%2C%22bootstrapScript%22%3A%22%2Fopt%2Fatom%2Fresources%2Fapp.asar%2Fsrc%2Fwindow-bootstrap.js%22%2C%22resourcePath%22%3A%22%2Fopt%2Fatom%2Fresources%2Fapp.asar%22%2C%22devMode%22%3Afalse%2C%22safeMode%22%3Afalse%2C%22appVersion%22%3A%221.0.19%22%2C%22shellLoadTime%22%3A1113%2C%22initialPaths%22%3A%5B%22%2Fhome%2Fmrx%2FDownloads%22%5D%7D:3

SyntaxError: Unexpected token /
  at Object.parse (native)
  at /home/mrx/.atom/packages/Remote-FTP/lib/client.js:78:17
  at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:380:3)

```
### Commands

```
     -1:52.6.0 remote-ftp:toggle (div.settings-view.pane-item)
     -0:56.4.0 core:select-right (atom-text-editor.editor.is-focused)
  2x -0:55.9.0 core:paste (atom-text-editor.editor.is-focused)
  2x -0:43.3.0 core:backspace (atom-text-editor.editor.is-focused)
  2x -0:37.8.0 core:save (atom-text-editor.editor.is-focused)
     -0:10.3.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:09.6.0 core:save (atom-text-editor.editor.is-focused)
```
### Config

``` json
{
  "core": {}
}
```
### Installed Packages

``` coffee
# User
Remote-FTP, v0.7.8

# Dev
No dev packages
```

**Comment by perfecto25** [comment]
running on mint, 3.16.0-38-generic #52~14.04.1-Ubuntu

**Comment by davidvandemaele** [comment]
I have exact the same error on Windows 7 64-bit

**Comment by sonic0002** [comment]
Same error happens for me

**Comment by szszoke** [comment]
I can confirm the existence of this bug in Windows 10 64 Bit, and in Ubuntu 15.04 64 Bit too.

**Comment by snettah** [comment]
in Osx yosemite 10.10.4

**Comment by jpxd** [comment]
There is an error parsing your `.ftpconfig` file...
Since the latest release we are pre-validating the JSON, so I don't understand why this error happens.
Can anybody of you post his `.ftpconfig` (without user&pw of course) please?

**Comment by snettah** [comment]
It's ok i'm sorry -_- , but I do not understand why we need to open a project for creates a file .ftpconfig, the button "create sftp file" not working if is no open project.
Thanks

**Comment by jpxd** [comment]
@snettah Remote-FTP is project based and creates a local copy of all files you edit on the remote server in your project directory on the disk (if you haven't already). Hence, a project directory is required.
But this is off-topic.

When the error `Uncaught SyntaxError: Unexpected token /` still occurs, post your .ftpconfig file (without credentials) here please.

**Comment by szszoke** [comment]
I don't have access to my pc but I've used the package's built in function to create the config file.

**Comment by GitMan1896** [comment]
I was having trouble with "Uncaught SyntaxError: Unexpected token /". I have removed all comments and now the FTP config example work like a charm!

**Comment by egemensoylu** [comment]
conf file copied from here > https://atom.io/packages/remote-ftp

{
    "protocol": "ftp",
    "host": "111.111.111.111", // string - The hostname or IP address of the FTP server. Default: 'localhost'
    "port": 21, // integer - The port of the FTP server. Default: 21
    "user": "user", // string - Username for authentication. Default: 'anonymous'
    "pass": "pass", // string - Password for authentication. Default: 'anonymous@'
    "promptForPass": false, // boolean - Set to true for enable password dialog. This will prevent from using cleartext password in this config. Default: false
    "remote": "/home/user/",
    "secure": false, // mixed - Set to true for both control and data connection encryption, 'control' for control connection encryption only, or 'implicit' for implicitly encrypted control connection (this mode is deprecated in modern times, but usually uses port 990) Default: false
    "secureOptions": null, // object - Additional options to be passed to tls.connect(). Default: (null) see http://nodejs.org/api/tls.html#tls_tls_connect_options_callback
    "connTimeout": 10000, // integer - How long (in milliseconds) to wait for the control connection to be established. Default: 10000
    "pasvTimeout": 10000, // integer - How long (in milliseconds) to wait for a PASV data connection to be established. Default: 10000
    "keepalive": 10000 // integer - How often (in milliseconds) to send a 'dummy' (NOOP) command to keep the connection alive. Default: 10000
}

**Atom Version**: 1.0.19
**System**: Ubuntu 14.04.3
**Thrown From**: [Remote-FTP](https://github.com/mgrenier/remote-ftp) package, v0.7.9

### Stack Trace

Uncaught SyntaxError: Unexpected token /

```
At file:///opt/atom/resources/app.asar/static/index.html > url params removed <

SyntaxError: Unexpected token /
  at Object.parse (native)
  at /home/egg/.atom/packages/Remote-FTP/lib/client.js:78:17
  at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:380:3)
```

**Comment by sktzofrenic** [comment]
{
    "protocol": "ftp",
    "host": "45.xx.xx.xxx", // string - The hostname or IP address of the FTP server. Default: 'localhost'
    "port": 21, // integer - The port of the FTP server. Default: 21
    "user": "root", // string - Username for authentication. Default: 'anonymous'
    "pass": "somepw", // string - Password for authentication. Default: 'anonymous@'
    "promptForPass": false, // boolean - Set to true for enable password dialog. This will prevent from using cleartext password in this config. Default: false
    "remote": "_myground",
    "secure": false, // mixed - Set to true for both control and data connection encryption, 'control' for control connection encryption only, or 'implicit' for implicitly encrypted control connection (this mode is deprecated in modern times, but usually uses port 990) Default: false
    "secureOptions": null, // object - Additional options to be passed to tls.connect(). Default: (null) see http://nodejs.org/api/tls.html#tls_tls_connect_options_callback
    "connTimeout": 10000, // integer - How long (in milliseconds) to wait for the control connection to be established. Default: 10000
    "pasvTimeout": 10000, // integer - How long (in milliseconds) to wait for a PASV data connection to be established. Default: 10000
    "keepalive": 10000 // integer - How often (in milliseconds) to send a 'dummy' (NOOP) command to keep the connection alive. Default: 10000
}

**Comment by jwpalmer** [comment]
For some reason (for me, at least), this error was getting triggered due to the comment ( // string - the... ) slashes. Removing all the comments has everything working just fine.

Not sure if this is due to the way the .ftpconfig file is being parsed or is somehow related to Atom, but either way everything works great now.

I also had to blow away my machine a few nights ago and had the exact same problem again, resolved the same way with a completely fresh environment. Couldn't remember how I did it so hoping this comment will at a minimum remind me in the future, but ideally help others.

Setup:
Mac OSX 10.11
Atom 1.1.0
Remote-FTP 0.7.8, 0.7.9

**Comment by jwpalmer** [comment]
Upon further review, I'm 99% confident the comments are causing the issue.

On `line 78, /lib/client.js`:
`JSON.parse` is called on the "streamed" .ftpconfig string, which is not to [spec for `JSON.parse`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse)

Here's a demonstration of what's happening so you can play with it a bit:

http://codepen.io/jwp/pen/QjxzKB

The tricky thing is, I'm not sure the best approach to resolve, aside from trying to write some massive function or half-baked regex to ignore/wipe the comments from the string.

At a minimum, it seems like a user-friendly error like "hey, remove the comments from the config and everything will work" would be a good approach to start.

**Comment by adamlukacka** [comment]
this config fike has worked for me

{
    "protocol": "ftp",
    "host": "xxx.xxx.xxx.xxx",
    "port": 21,
    "user": "xxx.xx",
    "pass": "xyzabcpoi",
    "promptForPass": false,
    "remote": "/",
    "secure": false,
    "secureOptions": null,
    "connTimeout": 10000,
    "pasvTimeout": 10000,
    "keepalive": 10000 
}

**Comment by jpxd** [comment]
In JSON you can't use comments and thats why parsing fails. But there is a module for that called `strip-json-comments` which now sanitizes the config file before parsing. Fixed in commit 2e8677f9857f92b103ca1180f6177f4ee24fd1b7
Simply remove the comments in .ftpconfig or update Remote-FTP to resolve the issue :smile:

**Comment by jwpalmer** [comment]
Woohoo, thanks, Jan!
